### PR TITLE
Using Docker instead of a package manager

### PR DESCRIPTION
**Pull request by roelbondoc**
Blog post about using docker for more than just for web development.

### CODE REVIEW

**Comment by roelbondoc** [comment]
@sammyd Hey Sam, could you please take a look at this when you get a chance? I'm looking for a tech review as well as any further input as to where you think I should expand on any of the things I touch on. Thanks!

File: _posts/2020-01-10-using-docker-instead-of-a-package-manager.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+author_bio: "Interests: Ruby on Rails, games, and basketball."
+author_image: "roel-bondoc@2x.jpeg"
+color: "#049474"
+hero: "c-post-hero--dark"
+image: "/assets/img/pattern-7@2x.png"
+category: "development"
+excerpt: "Keeping your computer clean with Docker."
+---
+
+Docker is a widely known containerization strategy. One of its main use is to provide developers with a consistent environment for building, packaging, and deploying applications. At raywenderlich.com, we use Docker containers to build our service oriented architecture. Most of our containers are based on the ruby:2.5-slim image (and other close variants). While there are many resources describing how Ruby on Rails projects can be setup and run using Docker, I want to write about using Docker in another fashion.
```
Comment (sammyd) [inline _posts/2020-01-10-using-docker-instead-of-a-package-manager.md:18]: could even say "widely used"

"One of its main uses..."

File: _posts/2020-01-10-using-docker-instead-of-a-package-manager.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+As it turns out, because of the popularity of Docker, it’s actually a great replacement for a package manager. If you are familiar with `aptitude` for Debian based Linux distributions, or `homebrew` for MacOS, then you know how useful they are for managing installed libraries and applications. What I dislike about package managers is how they directly modify your host system. Installing, upgrading, and removing packages never seem like a clean process to me. After working closely with Docker over a few years, I’ve learned how powerful and convenient it can be for normal day to day use and not just for developing applications.
```
Comment (sammyd) [inline _posts/2020-01-10-using-docker-instead-of-a-package-manager.md:20]: "never seems like a clean process"

**Review by sammyd — COMMENTED** [review]
Hi Roel,

Sorry for the delay in me taking a look at this.

I love the concept of this—something that now seems pretty obvious, but probably isn't to people new to docker.

I have a few simple suggestions:

- I personally prefer it if the majority of it is written in 2nd or 3rd person, with your own experience at the end. That's not to say what you've got isn't good—just a personal preference to recast your problems into ones the reader might be having.
- I'd be a little more specific about the issues—what happens without a ruby version manager / python version manager / node version manager? What possible solutions are there out there? What problems do they have? Then introduce docker as an alternative solution.
- I think I'd downplay the "clean computer" argument. Sure it's important, and it's obviously very important to you, but I think it's a difficult argument If that's the /only/ reason. Like, as long as all my stuff works, why should I care what the inside of my computer looks like? I should never have to care what the inside of my computer looks like.
- I'd quite like to see a slightly more detailed example... e.g. I have things like awscli, terraform, heroku, ffmpeg, etc... installed via home-brew. Maybe take some and show how I could use docker instead. Or (something I often do) how I could grab a dev project I want to try from GitHub, and test it out inside a docker container.

Just a few suggestions—feel free to take them or leave them. I think it's a good piece—nice length, and solid topic. Good work.

**Pull request closed without merging**